File: app/src/main/java/com/guuguo/android/pikacomic/net/https/TrustAllCerts.java
package com.guuguo.android.pikacomic.net.https;

import java.security.cert.X509Certificate;

import javax.net.ssl.X509TrustManager;

/**
 * guode 创造于 2017-05-01.
 * 项目 pika
 */
public class TrustAllCerts implements X509TrustManager {
    @Override
    public void checkClientTrusted(X509Certificate[] chain, String authType) {
    }

    @Override
    public void checkServerTrusted(X509Certificate[] chain, String authType) {
    }

    @Override
    public X509Certificate[] getAcceptedIssuers() {
        return new X509Certificate[0];
    }


}    

File: app/src/main/java/com/guuguo/android/pikacomic/net/https/TrustAllCerts.kt
package com.guuguo.android.pikacomic.net.https

import java.security.SecureRandom
import java.security.cert.CertificateException
import java.security.cert.X509Certificate
import javax.net.ssl.*

/**
 * Created by Yangmu on 17/1/19.
 */

class TrustAllCerts : X509TrustManager {
    @Throws(CertificateException::class)
    override fun checkClientTrusted(chain: Array<X509Certificate>, authType: String) {
    }

    @Throws(CertificateException::class)
    override fun checkServerTrusted(chain: Array<X509Certificate>, authType: String) {
    }

    override fun getAcceptedIssuers(): Array<X509Certificate?> {
        return arrayOfNulls(0)
    }

    class TrustAllHostnameVerifier : HostnameVerifier {
        override fun verify(hostname: String, session: SSLSession): Boolean {
            return true
        }
    }

    companion object {

        fun createSSLSocketFactory(): SSLSocketFactory? {
            var ssfFactory: SSLSocketFactory? = null

            try {
                val sc = SSLContext.getInstance("TLS")
                sc.init(null, arrayOf<TrustManager>(TrustAllCerts()), SecureRandom())

                ssfFactory = sc.socketFactory
            } catch (e: Exception) {
            }

            return ssfFactory
        }
    }
}